src/app/canvas/canvas.component.ts

```diff
@@ -7,25 +10,35 @@ import { Component, OnInit, ViewChild,ElementRef, AfterViewInit } from '@angular
 })
 export class CanvasComponent implements OnInit,AfterViewInit {
 
-  @ViewChild("canvas") mcanvas?: ElementRef<HTMLCanvasElement>;
-  context?: CanvasRenderingContext2D | null;
-  constructor() { }
+  @ViewChild("canvas") _mCanvas?: ElementRef<HTMLCanvasElement>;
+  
+  canvas!: fabric.Canvas;
+  constructor(private canvasService:CanvasCoreService) {
+    this.canvasService.shapeCreated.subscribe((object)=>{
+      this.drawObject(object);
+    })
+  }
 
   ngAfterViewInit(): void {
-    if(this.mcanvas!=null)
+    if(this._mCanvas!=null)
     {
-      this.context = this.mcanvas.nativeElement.getContext('2d');      
+      this._mCanvas.nativeElement.width  = this._mCanvas.nativeElement.offsetWidth;
+      this._mCanvas.nativeElement.height = this._mCanvas.nativeElement.offsetHeight;
+      this.canvas = new fabric.Canvas(this._mCanvas.nativeElement); 
```

Comment:
You can use either this or just pass the canvas id to initialise fabric

Reply:
Tried with canvas id.